src/game.cpp

```diff
@@ -4197,13 +4198,68 @@ Creature *game::is_hostile_within( int distance )
 }
 
 //get the fishable critters around and return these
-std::vector<monster *> game::get_fishable( int distance )
-{
+std::vector<monster *> game::get_fishable( int distance, const tripoint &fish_pos )
+{
+    // We're going to get the contiguous fishable terrain starting at
+    // the provided fishing location (e.g. where a line was cast or a fish
+    // trap was set), and then check whether or not fishable monsters are
+    // actually in those locations. This will help us ensure that we're
+    // getting our fish from the location that we're ACTUALLY fishing,
+    // rather than just somewhere in the vicinity.
+
+    std::unordered_set<tripoint> visited;
+    const int min_x = fish_pos.x - distance;
+    const int max_x = fish_pos.x + distance;
+    const int min_y = fish_pos.y - distance;
+    const int max_y = fish_pos.y + distance;
+
+    const auto get_fishable_terrain = [&]( tripoint starting_point,
+    std::unordered_set<tripoint> &fishable_terrain ) {
+        std::queue<tripoint> to_check;
```

Comment:
Please change to a std::vector, we've found recently that std::deque doesn't work the way we expect it to, it has wildly different implementations on different systems, ranging from a minimum allocation chunk of 4KB to a maximum allocation chunk of one element.  I can dig up the issue outlining this if you're interested.

I think the best bet for this behavior is:
```
std::vector<thing> vec;
vec.reserve( <expected size> );
size_t cursor = 0;
vec.push_back( first_thing );
while( cursor < vec.size() ) {
    const thing& = vec[ cursor];
    cursor++;
    ...
    // As new things are discovered
    vec.push_back( new_thing );
```

Reply:
Actually nevermind.  That solution is overboard for such a small search. 

The default deque-based queue is fine for this.